presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -84,10 +93,42 @@ public PlanNodeStatsEstimate filterStats(
             Session session,
             TypeProvider types)
     {
-        return new FilterExpressionStatsCalculatingVisitor(statsEstimate, session, types).process(predicate)
+        Expression simplifiedExpression = simplifyExpression(session, predicate, types);
+        return new FilterExpressionStatsCalculatingVisitor(statsEstimate, session, types)
+                .process(simplifiedExpression)
                 .orElseGet(() -> normalizer.normalize(filterStatsForUnknownExpression(statsEstimate), types));
     }
 
+    private Expression simplifyExpression(Session session, Expression predicate, TypeProvider types)
```

Comment:
i did this and it worked great except we lost ability to test NOT (`com.facebook.presto.cost.TestFilterStatsCalculator#testNotStats`) because `com.facebook.presto.sql.planner.iterative.rule.SimplifyExpressions#rewrite` pushes NOT into the comparisons.
This reuse is really a good idea, I like it. For now i added a TODO note, since I can't sacrifice the testability and don't want to block PR this longer.